src/dvc_objects/fs/callbacks.py

```diff
@@ -104,18 +120,26 @@ def absolute_update(self, value: int) -> None:
         return super().absolute_update(value)
 
     @classmethod
-    def as_callback(cls, maybe_callback: Optional["Callback"] = None) -> "Callback":
+    def as_callback(
+        cls, maybe_callback: Optional[fsspec.callbacks.Callback] = None
+    ) -> "Callback":
         if maybe_callback is None:
             return DEFAULT_CALLBACK
-        return maybe_callback
+        if isinstance(maybe_callback, Callback):
+            return maybe_callback
+        return _FsspecCallbackWrapper(maybe_callback)
 
     @classmethod
     def as_tqdm_callback(
```

Comment:
Should we move `as_tqdm_callback` and `as_callback` to the mixin as well?

Reply:
We could move `as_tqdm_callback`, but I think `as_callback` should stay here to override the default fsspec implementation